mycroft/skills/skill_loader.py

```diff
@@ -0,0 +1,144 @@
+import gc
+import os
+import sys
+from time import time
+
+
+from mycroft.configuration import Configuration
+from mycroft.messagebus import Message
+from mycroft.util.log import LOG
+from .core import create_skill_descriptor, load_skill
+
+
+def _get_last_modified_date(path):
+    """Get the last modified date of the most recently updated file in a path.
+
+    Exclude compiled python files, hidden directories and the settings.json
+    file.
+
+    Arguments:
+        path: skill directory to check
+
+    Returns:
+        int: time of last change
+    """
+    all_files = []
+    for root_dir, dirs, files in os.walk(path):
+        dirs[:] = [d for d in dirs if not d.startswith('.')]
+        for f in files:
+            ignore_file = (
+                    f.endswith('.pyc') or
+                    f == 'settings.json' or
+                    f.startswith('.') or
+                    f.endswith('.qmlc')
+            )
+            if not ignore_file:
+                all_files.append(os.path.join(root_dir, f))
+
+    # check files of interest in the skill root directory
+    return max(os.path.getmtime(f) for f in all_files)
+
+
+class SkillLoader:
+    def __init__(self, bus, skill_directory):
+        self.bus = bus
+        self.skill_directory = skill_directory
+        self.skill_id = os.path.basename(skill_directory)
+        self.load_attempted = False
+        self.loaded = False
+        self.last_modified = 0
+        self.last_loaded = 0
+        self.instance = None
+        self.active = True
+
+    @property
+    def config(self):
+        return Configuration.get()
```

Comment:
The concern I have with making the config an instance attribute is what happens when the config changes.  Accessing an instance attribute will not cause Configuration.get() to be called again so if the configuration did change, this process would never see those changes.  The property may not be the best mechanism here but I thought it important to call the get() method each time the config attribute is referenced in case config changes needed to be loaded.  Am I thinking about this right?

Reply:
the `Configuration.get()`returns a reference to the configuration dict, since the dict is a mutable object and we do not overwrite it after creation we just modify the existing one any reference will be updated automatically.

So the property is definitely not needed.

Reply:
Gotcha.  I misunderstood when you explained this to me the first time.  I will make it an instance attribute instead.